### PR TITLE
chore: Ensure all audit types in ResourceTable match APGL

### PR DESCRIPTION
**Pull request by Emyrk**
# What this does

We had a regression that a entry in APGL `audit.Auditable` did not have an entry in the enterprise `audit.AuditableResources`. 

Now `audit.AuditableResources` and `audit.Auditable` must match exactly.

# Additional Checks

- All exported fields have an audit action
- All fields with actions actually exist on the struct.

### CODE REVIEW

File: enterprise/audit/table.go [hunk 1]
```diff
@@ -41,17 +43,31 @@ const (
 // field's AuditType.
 type Table map[string]map[string]Action
 
+// Add adds a new entry to the table.
+func (t *Table) Add(key string, value map[string]Action) *Table {
```
Comment (kylecarbs) [inline enterprise/audit/table.go:47]: Might wanna make this a private func instead

File: enterprise/audit/table.go [hunk 2]
```diff
@@ -149,17 +165,50 @@ var AuditableResources = auditMap(map[any]map[string]Action{
...
 		"exp":         ActionTrack,
 		"uuid":        ActionTrack,
-	},
-})
+	}))
+
+// entry is a helper function that ensures all entries in the table are valid
+// audit.Auditable types. It also ensures all json tags have a corresponding
+// action.
+func entry[A audit.Auditable](v A, f map[string]Action) (string, map[string]Action) {
```
Comment (kylecarbs) [inline enterprise/audit/table.go:182]: Can we merge `entry` and `add` into one? That way it's not doubly nested

**Review by kylecarbs — APPROVED** [review]

File: enterprise/audit/table.go (hunk 2 above)
Comment (Emyrk) [reply enterprise/audit/table.go:180]: I am trying to play with the syntax still. This code does what I want, it checks enterprise matches APGL (but not the other way around...).


You cannot have a generic type constraint on a method (func attached to struct). So I have to make the `entry` function outside the `Table`. 

All generic type constraints are not available at runtime, so I cannot use `reflect` on `audit.Auditable`. I _can use `"golang.org/x/tools/go/packages"`_ and parse the APGL `audit` package.... which I might have to do to make sure the generic type constraint is fully represented in the table :cry:.

----

I am going to spend a little more time and see if I can make this a bit better syntax wise.

**Review by kylecarbs — APPROVED** [review]
Sick

**Pull request merged**